### ISSUE TITLE
SVGs Cut

### ISSUE THREAD

**Issue by mozzymoz** [iamdarrenhall/gulp-svg-spritesheet#3]
Hi there,

After several attempts to get the plugin to generate a sprite it finally worked, however not all SVGs are imported into the sprite and some are distorted. Any idea how I can fix this?

**Comment by iamdarrenhall** [comment]
It sounds like there could be issues with your SVGs. Check that your SVGs are well formed, and that they work as background images individually.

**Comment by mozzymoz** [comment]
Hi iamdarrenhall,

The SVGs alone work perfectly fine. I also ran them through svgo and svgmin without problems. 
These are quite complex shapes. Unfortunately I cannot post an example because of copyright issues (they are company logos).

**Comment by iamdarrenhall** [comment]
Is it something you can email me? My email address is on [my website](http://darrenhall.info/contact).

**Comment by mozzymoz** [comment]
Hey darren. I just sent you an email with the SVG in question. Thanks for taking the time to look into it. I really appreciate it.

**Comment by iamdarrenhall** [comment]
Thanks for sending me that! A quick look tells me that the SVG generated by the plugin does not include the xmlns:xlink="http://www.w3.org/1999/xlink" on the root SVG element, as a quick fix you can add it on line 67 of index.js.

```
var $ = cheerio.load('<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" />', { xmlMode: true }),
```

I'll look and see if there's better solution.

**Comment by mozzymoz** [comment]
Hey,
thanks for the suggestion. It seems that this was not the problem. I had a look at the generated SVG sprite and ran it through http://validator.w3.org/. What it found was that there are duplicate IDs in the sprite as most if not all of my SVGs define elements with IDs 'b' and 'a' etc. (generated by Illustrator).
I tried loading the SVG in question alone into the SVG and it renders fine.

Probably a good idea to replace duplicate ids in your module?

**Comment by mozzymoz** [comment]
Alright, I got it to work now :) My mistake was to run all the SVGs through svgo before running your task. So svgo replaces all IDs with single letters. This lead to duplicate IDs. The original SVGs actually have longer ID values

**Comment by iamdarrenhall** [comment]
Yeah, I'd run SVGO on the output rather than the input.

**Comment by mozzymoz** [comment]
So things are getting to where I want them. However now it is an issue of when to load which task. If I optimize before generating the sheet I get too many duplicated IDs, however when I run svgo or svgmin on the sprite it takes almost 4minutes to do the optimization (sheet size is 35MB).

I think I'll fork your repo over the weekend and try to find a solution for the duplicated ID issue. This one is going to be tricky.

**Comment by iamdarrenhall** [comment]
Your generated sprite sheet is 35MB?!

**Comment by iamdarrenhall** [comment]
Fixed by #4.
